### ISSUE TITLE
Not giving a name to to an uinput-sink raises an Exception or can even totally disconnect the keyboard 

### ISSUE THREAD

**Issue by mklcp** [david-janssen/kmonad#218]
So this first config works fine

1.
````
(defcfg
  input  (device-file "/dev/input/by-path/platform-i8042-serio-0-event-kbd")

  output (uinput-sink "?")

  fallthrough true
)

(defsrc
	spc
)


(deflayer base
	spc
)
````

This second one, where I just removed the name of the uinput-sink, triggers an infinite press of enter key as in #215.

2.
````
(defcfg
  input  (device-file "/dev/input/by-path/platform-i8042-serio-0-event-kbd")

  output (uinput-sink "")

  fallthrough true
)

(defsrc
	spc
)


(deflayer base
	spc
)
````

On other config, I interrupt this infinite press by pressing any key. Here, kmonad prints
````
Encountered error in KeySink: Could not encode Keyaction LinuxKeyEvent (1618923633,699069182,1,28,1) to bytes for writing to 
Encountered error in <emitter_proc>: Could not encode Keyaction LinuxKeyEvent (1618923633,699069182,1,28,1) to bytes for writing to 


kmonad: ExceptionInLinkedThread ThreadId 13 Could not encode Keyaction LinuxKeyEvent (1618923633,699069182,1,28,1) to bytes for writing to 
````

And now the worst one, where both the name of the sink and the fallthrough = true option are removed:

````
(defcfg
  input  (device-file "/dev/input/by-path/platform-i8042-serio-0-event-kbd")

  output (uinput-sink "")
)

(defsrc
	spc
)


(deflayer base
	spc
)
````

The infinite enter happens again; kmonad crashes, and my keyboard seems to be disconnected; I just can't control the computer. I have to manually shutdown it w/ the power button

**Comment by slotThe** [comment]
> The infinite enter happens again; kmonad crashes, and my keyboard seems to be disconnected; I just can't control the computer. I have to manually shutdown it w/ the power button

Had kmonad crashed you would have full control of your keyboard again, as the keyboard we create would also be destroyed.  What happens is that, because you've disabled the fallthrough but have not specified a full layout, we only grab a single key (space, in this case) and flat out ignore everything else.  Hence the keyboard becomes essentially unusable.

**Comment by mklcp** [comment]
Ok, so the only problem here is the exception message, which does not seem to cause any harm anyway
The infinite enter is another bug, and the incomplete keymap is my fault

**Comment by david-janssen** [comment]
Thanks for the issues.

I've fixed the forever-enter state here: #233

For the empty uinput name, good catch! I never thought of trying that. I've added a proper KMonad error, so this way at least we'll break as soon as possible with an informative warning. Should be alright now in:

197a4ef48e425ea78347905fe6c2151a6bf10830
